BitFaster.Caching/IScopedCache.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BitFaster.Caching
+{
+    /// <summary>
+    /// Represents a generic cache of key/scoped IDisposable value pairs.
+    /// </summary>
+    /// <typeparam name="K">The type of keys in the cache.</typeparam>
+    /// <typeparam name="V">The type of values in the cache.</typeparam>
+    public interface IScopedCache<K, V> where V : IDisposable
+    {
+        /// <summary>
+        /// Gets the total number of items that can be stored in the cache.
+        /// </summary>
+        int Capacity { get; }
+
+        /// <summary>
+        /// Gets the number of items currently held in the cache.
+        /// </summary>
+        int Count { get; }
+
+        /// <summary>
+        /// Attempts to create a lifetime for the value associated with the specified key from the cache
+        /// </summary>
+        /// <param name="key">The key of the value to get.</param>
+        /// <param name="lifetime">When this method returns, contains a lifetime for the object from the cache that has the specified key, or the default value of the type if the operation failed.</param>
+        /// <returns>true if the key was found in the cache; otherwise, false.</returns>
+        bool ScopedTryGet(K key, out Lifetime<V> lifetime);
```

Comment:
This is kind of a bad API because its cumbersome to wrap lifetime in a using statement.

Reply:
This can be provided as an extension method.